tests/test_transformations.py

```diff
@@ -1812,11 +1750,79 @@ def test_add_coordinate_system(
 
         assert csm.number_of_coordinate_systems == exp_num_cs
         if child_in_parent:
-            assert csm.get_local_coordinate_system(name, parent) == lcs
-            assert csm.get_local_coordinate_system(parent, name) == lcs.invert()
+            assert csm.get_cs(name, parent) == lcs
+            assert csm.get_cs(parent, name) == lcs.invert()
+        else:
+            assert csm.get_cs(name, parent) == lcs.invert()
+            assert csm.get_cs(parent, name) == lcs
+
+    # test_add_cs_reference_time -------------------------------------------------------
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "has_timestamp_csm, has_timestamp_lcs_1, has_timestamp_lcs_2, should_raise",
+        [
+            (True, False, False, False),
+            (True, True, False, False),
+            (True, False, True, False),
+            (True, True, True, False),
+            (False, False, False, False),
+            (False, True, False, True),
+            (False, False, True, True),
```

Comment:
```suggestion
            (False, True, False, Exception),
            (False, False, True, Exception),
```
I like this setup, maybe explicitly list the exception here directly for more flexibility (see below)